memguard.go

```diff
@@ -88,11 +98,18 @@ func New(length int, readOnly bool) (*LockedBuffer, error) {
 		b.MarkAsReadOnly()
 	}
 
-	// Append this LockedBuffer to allLockedBuffers.
+	// Append the lockedBuffer to allLockedBuffers. We have to add lockedBuffer
+	// instead of LockedBuffer so that the finalizerGuard can become unreachable
 	allLockedBuffersMutex.Lock()
-	allLockedBuffers = append(allLockedBuffers, b)
+	allLockedBuffers = append(allLockedBuffers, ib)
 	allLockedBuffersMutex.Unlock()
 
+	// Use a finalizer to make sure the buffer gets destroyed even if the user
+	// forgets to do it
+	runtime.SetFinalizer(b.finalizerGuard, func(_ *finalizerGuard) {
```

Comment:
the function passed to SetFinalizer need to have a single argument and no return value. This single argument must be able to be assigned the first argument to SetFinalizer (b.finalizerGuard). Since b.finalizerGuard is *finalizerGuard, the simplest way to meet this requirement is to make the single argument of the function be of type *finalizerGuard (but it could also be something else, e.g. interface{}).

Since we don't need the guard itself (the only purpose of the guard is to know when LockedBuffer has become unreachable: when LockedBuffer becomes unreachable also finalizerGuard becomes unreachable), we ignore the argument (hence _)